exla/lib/exla/backend.ex

```diff
@@ -0,0 +1,268 @@
+defmodule EXLA.Backend do
+  @moduledoc """
+  A Nx tensor backend for the data kept on the device.
+
+  You can directly transfer to this backend by calling
+  `Nx.backend_transfer/2` or `Nx.backend_copy/2`. It
+  allows the following options:
+
+    * `:client` - the client to store the data on.
+      Defaults to the client configured in `Nx.Defn`,
+      otherwise uses `:host`.
+
+    * `:device_id` - which device to store it on
+
+  To get the data out of the device backend into a regular
+  tensor, call `Nx.backend_transfer/1` (with the device
+  tensor as the single argument).
+  """
+
+  @behaviour Nx.Backend
+  @enforce_keys [:buffer]
+  defstruct [:buffer]
+
+  alias Nx.Tensor, as: T
+  alias EXLA.Backend, as: B
+
+  import Nx.Shared
```

Comment:
This is private Nx API and we should not rely on it. Maybe you can try calling `Nx.BinaryBackend.constant() |> Nx.to_binary()` or something like that?

Reply:
Or create on binary backend and transfer.

Reply:
Did `Nx.BinaryBackend.constant` + `Nx.BinaryBackend.backend_transfer`. Not sure if it's better to call `backend_transfer` on Nx or directly on the backend, but the latter seems more consistent given we call `constant` directly.